Sources/MuxUploadSDK/Public API/UploadManager.swift

```diff
@@ -89,18 +77,53 @@ public final class UploadManager {
         }
     }
     
+    /// Adds an ``UploadsUpdatedDelegate`` with the given ID. You can add as many of these as you like, each with a uniqe ID
+    public func addUploadsUpdatedDelegate(id: Int, _ delegate: UploadsUpdatedDelegate?) {
```

Comment:
What's the purpose of the `id` parameter here?

Reply:
`UploadsUpdatedDelegate` isn't hashable, and I need to store it in a Set (so it can later be removed from the Set of delegates). I chose to do this by token, as I can't see any way to enforce compliance with `Hashable` on the `UploadsUpdatedDelegate`

Reply:
Alternatively the protocol can [require `UploadsUpdatedDelegate` be a reference type](https://developer.apple.com/documentation/swift/anyobject) instead of requiring it to be `Hashable`. Internally the SDK can then use [`ObjectIdentifier`](https://developer.apple.com/documentation/swift/objectidentifier) as a unique ID for each delegate instance.

```
protocol UploadsUpdatedDelegate: AnyObject
```

Either `Hashable` or `AnyObject` would work here, the latter is closer to what UIKit and other Cocoa APIs require of delegates.